src/modules/local_position_estimator/sensors/landing_target.cpp

```diff
@@ -0,0 +1,128 @@
+#include "../BlockLocalPositionEstimator.hpp"
+#include <systemlib/mavlink_log.h>
+#include <matrix/math.hpp>
+
+extern orb_advert_t mavlink_log_pub;
+
+static const uint64_t 	TARGET_TIMEOUT =   2000000; // [us]
+
+void BlockLocalPositionEstimator::landingTargetInit()
+{
+	if (_target_mode.get() == Target_Moving) {
+		// target is in moving mode, do not initialize
+		return;
+	}
+
+	Vector<float, n_y_target> y;
+
+	if (landingTargetMeasure(y) == OK) {
+		mavlink_and_console_log_info(&mavlink_log_pub, "Landing target init");
+		_targetInitialized = true;
+	}
+}
+
+int BlockLocalPositionEstimator::landingTargetMeasure(Vector<float, n_y_target> &y)
+{
+	if (_target_mode.get() == Target_Stationary) {
+		if (_sub_landing_target_pose.get().rel_vel_valid) {
+			y(0) = _sub_landing_target_pose.get().vx_rel;
```

Comment:
have you considered fusing the position, besides the velocity?

Reply:
I have considered fusing position and defining the beacon's position via parameter. There are some problems with that because you have to deal with the discrepancy between the position reported by the GPS and the one reported by the target. This is not needed when fusing velocity.

Reply:
Ok that makes sense.